### PR TITLE
Upload profile photo from certain path

### LINKED ISSUE #31: Add command to upload display picture

### LINKED ISSUE #77: Display picture

### LINKED ISSUE #83: As a student, I want to upload a photo of myself or my teammates
so that I can help me to recognize my teammates.

#77

### LINKED ISSUE #138: Add functionality to display the profile photo for each user

### PR DESCRIPTION
**Pull request by Livian1107**
close #139 
close #138 
close #31 
close #83 
close #77

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -310,6 +312,7 @@ Examples:
 * `theme`
 * `t`
 
+// tag::undoredo[]
```
Comment (kohyeeru) [inline docs/UserGuide.adoc]: Can anyone help to confirm if having multiple instances of the same tag in the guide will result in any linking issues (perhaps from the personal portfolio for starters?)

File: src/main/java/seedu/progresschecker/logic/commands/UploadCommand.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        requireNonNull(personToUpdate);
+        try {
+            model.uploadPhoto(personToUpdate, toUpload);
+            return new CommandResult(MESSAGE_SUCCESS);
+        } catch (IOException e) {
+            throw new CommandException(MESSAGE_IMAGE_NOT_FOUND);
+        } catch (DuplicatePersonException e) {
+            throw new CommandException(MESSAGE_IMAGE_DUPLICATE);
+        } catch (PersonNotFoundException pnfe) {
+            throw new AssertionError("The target person cannot be missing");
```
Comment (kohyeeru) [inline src/main/java/seedu/progresschecker/logic/commands/UploadCommand.java:57]: A new constant variable can be created to be used here e.g. ```CommandResult(MESSAGE_PERSON_NOT_FOUND)```.

File: src/main/java/seedu/progresschecker/model/person/Person.java [hunk 3]
```diff
@@ -15,12 +15,15 @@
...
 
+    private static String defaultPath = "/images/profile_photo.jpg";
+
     private final Name name;
     private final Phone phone;
     private final Email email;
     private final GithubUsername username;
     private final Major major;
     private final Year year;
+    private String photoPath;
```
Comment (kohyeeru) [inline src/main/java/seedu/progresschecker/model/person/Person.java:26]: Would it be better to create a new ```PhotoPath.java``` class for this?

File: src/main/java/seedu/progresschecker/ui/PersonCard.java [hunk 4]
```diff
@@ -2,20 +2,24 @@
...
+import javafx.scene.paint.ImagePattern;
+import javafx.scene.shape.Rectangle;
+import seedu.progresschecker.MainApp;
 import seedu.progresschecker.model.person.Person;
 
 /**
  * An UI component that displays information of a {@code Person}.
  */
-public class PersonCard extends UiPart<Region> {
+public class  PersonCard extends UiPart<Region> {
```
Comment (kohyeeru) [inline src/main/java/seedu/progresschecker/ui/PersonCard.java]: Additional space here? o:

File: src/test/java/seedu/progresschecker/logic/commands/UploadCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import seedu.progresschecker.logic.CommandHistory;
+import seedu.progresschecker.logic.UndoRedoStack;
+import seedu.progresschecker.model.Model;
+import seedu.progresschecker.model.ModelManager;
+import seedu.progresschecker.model.ProgressChecker;
+import seedu.progresschecker.model.UserPrefs;
+import seedu.progresschecker.model.person.Person;
+
+/**
+ * Contains integration tests (interaction with the Model, UndoCommand and RedoCommand) and unit tests for EditCommand.
```
Comment (kohyeeru) [inline src/test/java/seedu/progresschecker/logic/commands/UploadCommandTest.java]: Forgot to update the header comment?

**Review by kohyeeru — CHANGES_REQUESTED** [review]
Looking good. This review only touches on code consistency and not its functionality. I'll leave the review of confirmation that this is working to someone else.

**Review by AdityaA1998 — APPROVED** [review]
Looks fine to me

**Review by kohyeeru — APPROVED** [review]

**Pull request merged**